common-auth/SECURITY.md

```diff
@@ -0,0 +1,65 @@
+hadoop-auth module and having more control over the changes needed in future. Some text for this document has been borrowed from SECURITY.md of Apache Storm  
+
+By default, registry and streamline web-services are running without authentication enabled and anyone can access the web-services from ui/client as far as they
+know the url and can access the web-server from the client machine. To enable authentication with the client, webservice needs to add a servlet filter from 
+this module. The webservice module will need to declare a dependency on this module. One way of adding a servlet filter in code is as follows. 
+
+```java
+List<ServletFilterConfiguration> servletFilterConfigurations = registryConfiguration.getServletFilters();
+if (servletFilterConfigurations != null && !servletFilterConfigurations.isEmpty()) {
+    for (ServletFilterConfiguration servletFilterConfiguration: servletFilterConfigurations) {
+        try {
+            FilterRegistration.Dynamic dynamic = environment.servlets().addFilter(servletFilterConfiguration.getClassName(), (Class<? extends Filter>)
+            Class.forName(servletFilterConfiguration.getClassName()));
+            dynamic.setInitParameters(servletFilterConfiguration.getParams());
+            dynamic.addMappingForUrlPatterns(EnumSet.allOf(DispatcherType.class), true, "/*");
+        } catch (Exception e) {
+            LOG.error("Error registering servlet filter {}", servletFilterConfiguration);
+            throw new RuntimeException(e);
+        }
+    }
+}
+```
+
+ServletFilterConfiguration is a java object representing any servlet filter used in schema registry. However the general idea is that one needs to
+add com.hortonworks.registries.auth.server.AuthenticationFilter for enabling authentication 
+
+The configuration for that filter in yaml is represented in params property as follows.
+
+```yaml
+servletFilters:
+ - className: "com.hortonworks.registries.auth.server.AuthenticationFilter"
+   params:
+     type: "kerberos"
+     kerberos.principal: "HTTP/web-service-host.com"
+     kerberos.keytab: "/path/to/keytab"
+     kerberos.name.rules: "RULE:[2:$1@$0]([jt]t@.*EXAMPLE.COM)s/.*/$MAPRED_USER/ RULE:[2:$1@$0]([nd]n@.*EXAMPLE.COM)s/.*/$HDFS_USER/DEFAULT"
+     token.validity: 36000
+```
+
+The servlet filter uses the principal `HTTP/{hostname}` (here hostname should be the host where the web-service runs) to login. Make sure that principal is 
+created as part of Kerberos setup
+
+Once configured, you must do `kinit` on client side using the principal of the user before accessing the web-service via the browser or some other client. 
+This principal also needs to be created first during Kerberos setup
+
+Here's an example of accessing web-service after the setup above:
+```bash
+curl  -i --negotiate -u:anyUser  -b ~/cookiejar.txt -c ~/cookiejar.txt  http://<web-service-host>:<port>/api/v1/
```

Comment:
It could be. How do you want me to word it? We described above what needs to be done on web server side and client side from Kerberos point of view and then we are just giving an example of a client aka curl. Even the settings change in browser is not clear and confusing. I did not understand when i read it in the first place. I think its out of the scope trying to explain documentation about browser settings or curl